src/test/suite/demoTutorials.test.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Uri, commands }  from 'vscode';
+import { ensureExtensionActivated, validateCommands, getFailedCommands } from './Utils';
+import { expect } from 'chai';
+import * as extensionFunctions from '../../extensionFunctions';
+import { beforeEach } from 'mocha';
+
+suite('Test the demo tutorials', () => {
+
+	const filesToTest : string[] = [
+		`demos/markdown/dep-table.didact.md`,
+		`demos/markdown/didact-demo.didact.md`,
+		`demos/markdown/simple-example.didact.md`,
+		`demos/markdown/helloJS/helloJS.didact.md`,
+		`demos/markdown/tutorial/tutorial.didact.md`,
+		`demos/asciidoc/simple-example.didact.adoc`,
+		`demos/asciidoc/didact-demo.didact.adoc`,
+		`demos/asciidoc/dep-table.didact.adoc`,
+		`create_extension/create-new-tutorial-with-extension.didact.md`
+	];
+
+	setup(async () => {
+		ensureExtensionActivated();
+		return;
+	});
+
+	beforeEach(async () => {
+		await commands.executeCommand('workbench.action.closeAllEditors');
+		return;
+	});
+
+	test('Walk through the demo files to ensure that all commands exist in the VS Code system', async () => {
+		suite('walk through a list of demo files', () => {
+			filesToTest.forEach(function(fileToTest: string) {
+
+				beforeEach(async () => {
+					await commands.executeCommand('workbench.action.closeAllEditors');
+					return;
+				});
+			
+				test(`test provided demo file "${fileToTest}"`, async () => {
+					const testCommand = Uri.parse(`vscode://redhat.vscode-didact?extension=${fileToTest}`);
+					const isOk = await validateCommands(testCommand);
+					expect(isOk).to.be.true;
+				});
+			});
+		});
+	});
+
+	test('Walk through a tutorial we know has one command that fails due to vscode-java not being loaded', async() => {
+		const testCommand = `vscode://redhat.vscode-didact?extension=demos/markdown/camelinaction/chapter1/cia2-chapter-1-v2.didact.md`;
+		const isOk = await validateCommands(Uri.parse(testCommand));
+		expect(isOk).to.be.false;
+		const commands : any[] = extensionFunctions.gatherAllCommandsLinks();
+		const failedCommands = await getFailedCommands(commands);
+		expect(failedCommands).to.include('java.projectConfiguration.update');
+	});
```

Comment:
what about creating a dedicated file with an unknown command?

To validate this other tutorial, we will need to install the Java extension durign test which will break this test.

Reply:
The very next test for "vscode://redhat.vscode-didact?extension=demos/markdown/validation-test.didact.md" does exactly that. The test in this case is still useful and bubbles up the idea that there may be some tutorials depending on extensions outside of the vscode and Didact APIs. So I'm not sure why we would need to create yet another test?

Reply:
not realized the other test.

Then, I do not really see the value of this one. It will need to be removed when the test for this tutoiral will be implemented. And there is no difference if it is a completely "made-up" command or one from an extension not installed.

Reply:
If you look at the camel example, it approaches things from a VERY different point of view than the rest of the examples we've provided. I based it on the Camel In Action book, following the same pattern as expressed there. I see the value as it stretches the Didact framework in a different way.

Keep in mind this is a very simple set of tests. I'm working on adding some mocking capabilities so we can actually walk through a tutorial top to bottom to ensure that we're getting expected responses from things, including the command line in a terminal. So this is just the beginning.

Reply:
the scope of the test is to check if the commands provided exists or not. For this scope, I do not see differences between tutorials.

I do not understand what mocking comes into play to test a tutorial. What would be tested if the command called by didact is mocked?

Reply:
Mocking does not come in this particular code change, so don't worry about it. I'm still playing with a few things. 

And your concern with the tests is noted, but I'm leaving it as-is. Thanks.